crates/cli/src/new.rs

```diff
@@ -224,7 +224,10 @@ fn create_new_project(arg: NewArg) -> Result<()> {
 
 fn default_rule(id: &str, lang: SgLang) -> String {
   format!(
-    r#"id: {id}
+    r#"
+    # yaml-language-server: $schema=https://raw.githubusercontent.com/ast-grep/ast-grep/main/schemas/rule.json
```

Comment:
This is better than nothing, but unfortunately it does not handle schema changes between different ast-grep versions particularly well. I am not sure there is much we can do about this since we might always run a rule against a different ast-grep version than it was created for.

Reply:
I think this is good enough at the moment. We can release it and garner more feedbacks from users.